kam should trim leading and trailing spaces from input

**Describe the bug**
kam exits with error if the user provides answers with leading spaces in their response.  Trailing spaces can pose problems as well.

**To Reproduce**
Steps to reproduce the behavior:
1. Run kam bootstrap
2. When prompted to enter the URL for your GitOps repository, purposely add a space before your gitops URL
3. Hit enter

kam exits with an error

See output below:

```
kam bootstrap
  
Checking dependencies

 ✓  Checking if Sealed Secrets is installed with the default configuration [325ms]
 ✓  Checking if ArgoCD Operator is installed with the default configuration [721ms]
 ✓  Checking if OpenShift Pipelines Operator is installed with the default configuration [846ms]
  
Starting interactive prompt

? Provide the URL for your GitOps repository  https://github.com/keithchong/gitops9.git
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1fadcb7]

goroutine 1 [running]:
github.com/redhat-developer/kam/pkg/cmd/ui.EnterGitRepo(0xc0002ed820, 0x2659cf6)
	/Users/kchong/wrh/git/kam/pkg/cmd/ui/ui.go:26 +0x117
github.com/redhat-developer/kam/pkg/cmd.initiateInteractiveMode(0xc00007fd50, 0xc0002ed820, 0x2874620, 0xc000495500)
	/Users/kchong/wrh/git/kam/pkg/cmd/bootstrap.go:161 +0x9d
github.com/redhat-developer/kam/pkg/cmd.(*BootstrapParameters).Complete(0xc00007fd50, 0x265fca6, 0x9, 0xc0004fe2c0, 0x34268e8, 0x0, 0x0, 0xc00018fc00, 0x2659e0a)
	/Users/kchong/wrh/git/kam/pkg/cmd/bootstrap.go:112 +0x14c
github.com/redhat-developer/kam/pkg/cmd/genericclioptions.GenericRun(0x2874660, 0xc00007fd50, 0xc0004fe2c0, 0x34268e8, 0x0, 0x0)
	/Users/kchong/wrh/git/kam/pkg/cmd/genericclioptions/runnable.go:23 +0x6e
github.com/redhat-developer/kam/pkg/cmd.NewCmdBootstrap.func1(0xc0004fe2c0, 0x34268e8, 0x0, 0x0)
	/Users/kchong/wrh/git/kam/pkg/cmd/bootstrap.go:324 +0x5e
github.com/spf13/cobra.(*Command).execute(0xc0004fe2c0, 0x34268e8, 0x0, 0x0, 0xc0004fe2c0, 0x34268e8)
	/Users/kchong/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004fe000, 0x1008285, 0xc0000ae058, 0x0)
	/Users/kchong/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/kchong/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/redhat-developer/kam/pkg/cmd.Execute()
	/Users/kchong/wrh/git/kam/pkg/cmd/kam.go:42 +0x2a
main.main()
	/Users/kchong/wrh/git/kam/cmd/kam/kam.go:8 +0x25

```

**Expected behavior**
kam should trim spaces. I discovered this problem when I copied and pasted the repo, and I inadvertantly copied a leading space.

--- a/pkg/cmd/ui/ui.go
+++ b/pkg/cmd/ui/ui.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"net/url"
 	"path/filepath"
+	"strings"
 
 	"github.com/redhat-developer/kam/pkg/pipelines/ioutils"
 	"gopkg.in/AlecAivazis/survey.v1"
@@ -20,7 +21,7 @@ func EnterGitRepo() string {
 	}
 	err := survey.AskOne(prompt, &gitOpsURL, survey.Required)
 	handleError(err)
-
+	gitOpsURL = strings.TrimSpace(gitOpsURL)
 	p, err := url.Parse(gitOpsURL)
 	handleError(err)
 	if p.Host == "" {
@@ -53,7 +54,7 @@ func EnterImageRepoInternalRegistry() string {
 
 	err := survey.AskOne(prompt, &imageRepo, survey.Required)
 	handleError(err)
-	return imageRepo
+	return strings.TrimSpace(imageRepo)
 }
 
 // EnterDockercfg allows the user to specify the path to the docker config json
@@ -68,7 +69,7 @@ func EnterDockercfg() string {
 
 	err := survey.AskOne(prompt, &dockerCfg, nil)
 	handleError(err)
-	return dockerCfg
+	return strings.TrimSpace(dockerCfg)
 }
 
 // EnterImageRepoExternalRepository allows the user to specify the type of image
@@ -82,7 +83,7 @@ func EnterImageRepoExternalRepository() string {
 
 	err := survey.AskOne(prompt, &imageRepoExt, survey.Required)
 	handleError(err)
-	return imageRepoExt
+	return strings.TrimSpace(imageRepoExt)
 }
 
 // VerifyOutputPath allows the user to specify the path where the gitops configuration must reside locally in a UI prompt.
@@ -95,6 +96,7 @@ func VerifyOutputPath(originalPath string, overwrite, outputPathOverridden, prom
 	}
 	if !outputPathOverridden && promptForPath {
 		handleError(survey.AskOne(prompt, &outputPath, nil))
+		outputPath = strings.TrimSpace(outputPath)
 	}
 	for true {
 		exists, err := ioutils.IsExisting(ioutils.NewFilesystem(), filepath.Join(outputPath, "pipelines.yaml"))
@@ -107,6 +109,7 @@ func VerifyOutputPath(originalPath string, overwrite, outputPathOverridden, prom
 			break
 		}
 		handleError(survey.AskOne(prompt, &outputPath, nil))
+		outputPath = strings.TrimSpace(outputPath)
 	}
 	return outputPath
 }
@@ -147,7 +150,7 @@ func EnterSealedSecretService(sealedSecretService *types.NamespacedName) string 
 
 	err := survey.AskOne(prompt, &sealedNs, makeSealedSecretsService(sealedSecretService))
 	handleError(err)
-	return sealedNs
+	return strings.TrimSpace(sealedNs)
 }
 
 // EnterGitHostAccessToken , it becomes necessary to add the personal access
@@ -172,7 +175,7 @@ func EnterPrefix() string {
 	}
 	err := survey.AskOne(prompt, &prefix, makePrefixValidator())
 	handleError(err)
-	return prefix
+	return strings.TrimSpace(prefix)
 }
 
 // EnterServiceRepoURL , allows users to differentiate between the bootstrap and init options, addition of the service repo url will allow users to bootstrap an environment through the UI prompt.
@@ -184,7 +187,7 @@ func EnterServiceRepoURL() string {
 	}
 	err := survey.AskOne(prompt, &serviceRepo, survey.Required)
 	handleError(err)
-
+	serviceRepo = strings.TrimSpace(serviceRepo)
 	p, err := url.Parse(serviceRepo)
 	handleError(err)
 	if p.Host == "" {

